conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -60,12 +60,22 @@ def _get_requires_names(self, name, cpp_info):
         Note: CppInfo could be coming from one Component object instead of the dependency
         """
         ret = []
-        for req in cpp_info.requires:
-            pkg_name, comp_name = req.split("::") if "::" in req else (name, req)
-            # FIXME: it could allow defining requires to not direct dependencies
-            req_conanfile = self._conanfile.dependencies.host[pkg_name]
-            comp_alias_name = get_component_name(req_conanfile, comp_name)
-            ret.append(self._get_pc_name(pkg_name, comp_alias_name))
+        if cpp_info.requires:
```

Comment:
But ``.requires`` is mandatory and should be correctly defined for this to work. Is this an issue that the original package didn't declare ``requires``?

Reply:
It's mandatory but its value could be `[]`, right?

Reply:
But if it is ``[]`` it means that this component or thing doesn't depend on any transitive things, and as such, the generated ``.pc`` (or .cmake or whatever), will not transitively depend on anything.

Reply:
Then, if I have this scheme:

```
libA->libB->libC

libA requires libB
libB requires libC
```
why is the result of `cpp_info.requires` equal to `[]` when that `cpp_info` belongs to `libB`?